blocks/library/text/index.js

```diff
@@ -31,13 +31,29 @@ registerBlockType( 'core/text', {
 		content: query( 'p', children() ),
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'raw',
+				matcher: ( node ) => (
+					node.nodeName === 'P' &&
```

Comment:
Not sure why we should treat pasting and legacy content differently. IMO we should avoid losing data (means we should match if the classnames are known but not if it uses custom classnames or other attributes).

That said, I understand than this PR puts in place the base architecture for the pasting and legacy content feature and I'm fine tweaking those over time.